src/sas/sascalc/calculator/sas_gen.py

```diff
@@ -1117,12 +1123,17 @@ def read(self, path):
                             msg += "We accept only mT or A/m as valueunit"
                             logging.error(msg)    
                             return None
+                        elif valueunit.count("mT") > 1 or valueunit.count("A/m") > 1:    
+                            valueunit = valueunit.split(" ", 1)
+                            valueunit = valueunit[0].lstrip()
                     if s_line[0].lower().count("valuemultiplier") > 0:
                         valuemultiplier = s_line[1].lstrip()
-                    if s_line[0].lower().count("valuerangeminmag") > 0:
-                        valuerangeminmag = s_line[1].lstrip()
-                    if s_line[0].lower().count("valuerangemaxmag") > 0:
-                        valuerangemaxmag = s_line[1].lstrip()
+                    else: 
+                        valuemultiplier = 1
+                    # if s_line[0].lower().count("valuerangeminmag") > 0:
```

Comment:
If it isn't used anymore, I suggest getting rid of the code. If it might be used again in the future, document why it is currently commented out.